### DISCUSSION TITLE
Foxglove agent glob pattern

### DISCUSSION THREAD

**Discussion by hck007** [foxglove/community#1047] [Q&A]
I'm using the default storage path of the foxglove agent to automatically upload data. I have multiple subdirectories under the `STORAGE_ROOT` defined in `/etc/foxglove/agent/envfile`. For each sub directories, we have multiple layers of directories. Since `WATCH_AUTO_IMPORT_PATTERN` only supports two kinds of glob patterns * and ?. Is there a way to recursively automatically upload any number of layers of directories? If not, can we have foxglove agent upload up to 3 layers of directories e.g. `WATCH_AUTO_IMPORT_PATTERN="*.bag and */*.bag and */*/*.bag abd */*/*/*.bag"`

**Comment by james-rms** [comment]
Hi - we don't support recursive globs right now, though we may in the near future.
